File: nodejs/muse/data_structures/ArrayQueueTest.js
'use strict'

const ArrayQueue = require('muse/data_structures/ArrayQueue')
const TestRunner = require('muse/util/TestRunner')

const testArrayQueue = () => {
  const size = 8192

  const queue = new ArrayQueue.ArrayQueue(0)

  for (let i = 1; i <= size; i++) {
    queue.enqueue(i)
  }

  queue.shrink()

  if (queue.size() !== size) {
    return false
  }

  if (queue.capacity() !== size) {
    return false
  }

  for (let i = 1; i <= size; i++) {
    if (queue.peek() !== i) {
      return false
    }
    queue.dequeue()
  }

  queue.shrink()

  if (!queue.isEmpty()) {
    return false
  }

  return queue.capacity() === 0
}

if (module === require.main) {
  TestRunner.pick(testArrayQueue)
}

File: ruby/muse/data_structures/ArrayQueueTest.rb
require "muse/data_structures/ArrayQueue"
require "muse/util/TestRunner"

def testArrayQueue
  size = 8192

  queue = ArrayQueue::ArrayQueue.new(0)

  1.upto(size) do |i|
    queue.enqueue(i)
  end

  queue.shrink

  return false if queue.size != size

  return false if queue.capacity != size

  1.upto(size) do |i|
    return false if queue.peek != i

    queue.dequeue
  end

  queue.shrink

  return false unless queue.isEmpty

  queue.capacity.zero?
end

TestRunner.pick(testArrayQueue) if __FILE__ == $PROGRAM_NAME
